conda/recipes/libcudf_kafka/meta.yaml

```diff
@@ -25,8 +26,8 @@ requirements:
   build:
     - cmake >=3.20.1
   host:
-    - libcudf {{ version }}
-    - librdkafka >=1.5.0,<1.5.3
+    - libcudf {{version}} cuda{{ cuda_version }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
```

Comment:
Need this kind of pinning to build number level so that we near relevant errors like this: https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-cuda-build/CUDA=11.2/2592/console, else it would take quite a bit of time figuring out why conda is behaving weirdly here.

Reply:
This is totally fine. We mostly only use the core logic in librdkafka and that really doesn't even change between version releases.

Reply:
I don't think we need the libcudf pinning changes here, only the `librdkafka` changes. The conflict seems to be with `lz4-c` where `librdkafka 1.5.x` requires 1.9.2.x and `arrow-cpp 4.0.1` requires `1.9.3.x`.

Reply:
> I don't think we need the libcudf pinning changes here, only the `librdkafka` changes. The conflict seems to be with `lz4-c` where `librdkafka 1.5.x` requires 1.9.2.x and `arrow-cpp 4.0.1` requires `1.9.3.x`.

Correct, so the reason I had to add `libcudf` pinning here is `conda` was fetching `libcudf` packages from `nighlies` that were having `arrow-cpp-1.0.1` instead of `libcudf` that is available in `conda-build` dir. This was hard to find until I have added this pinning, with only this pinning conda started throwing the errors like in this [log](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-cuda-build/CUDA=11.2/2592/console) that helped understand what was wrong. Hence thought to retain this pinning as it would be helpful in future upgrades aswell.

Reply:
I think this will break release builds though since the release builds versioning doesn't use the git describe info.

Reply:
Is there a way to force conda to pick libcudf only from local conda build instead of nightly? 

cc: @ajschmidt8 @jakirkham

Reply:
Conda treats local builds as living in a channel with highest priority. So this should already be the case

If that doesn't work for some reason, we might want to use strict channel priority. There use to be some issues around that with labels, but I don't think those apply any more (since we don't label packages based on CUDA version any more). Though feel free to correct me if that is not the case

One thing I'm less clear on is how Project Flash would fit into all of this. AIUI that downloads cached packages from a server somewhere. How they are downloaded, whether they live in some kind of channel, and how they are installed? These are all interesting questions that I don't know the answer to (though AJ likely would)

Reply:
So conda here has access to both local and nightlies channel. Local has a libcudf conda build with arrow-4.0.1 and nightlies has a libcudf conda build with arrow-1.0.1. Because there was a dependency with previous version of librdkafka conda silently picked the libcudf conda build from nightlies channel. Can we avoid this if we enable strict channel priority?

Reply:
Yeah exactly that's what strict channel priority does. It only grabs packages from the highest priority channel. Though it may reveal some conflict or something that we are missing, which is causing the older `libcudf` to be picked up. We can either [configure it globally]( https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html#strict-channel-priority ) or [pass it to individual commands like `conda install`]( https://docs.conda.io/projects/conda/en/latest/commands/install.html?highlight=--strict-channel-priorit#Solver%20Mode%20Modifiers ).

Reply:
Had a discussion offline with ops and realized we have older version packages in higher priority channels. So realized both pinning `libcudf` and using `strict` channel priority can't be helpful in this scenario. Hence reverting it.